### DISCUSSION TITLE
RISCV Vector code problem

### DISCUSSION THREAD

**Discussion by huangxiansheng123** [gem5/gem5#2175] [Q&A]
![20240725-141035](https://github.com/user-attachments/assets/34fe9e22-499f-42bc-8790-cf658fec3b83)

Here on line 128 elen might be rewritten as vlen？

**Comment by huangxiansheng123** [comment]
The path of the file is src/arch/riscv/isa/formats/vector_conf.isa

**Comment by hnpl** [comment]
I believe that condition corresponds to this part of the spec,

https://github.com/riscv/riscv-v-spec/blob/master/v-spec.adoc?plain=1#L330

> The use of vtype encodings with LMUL < SEW_{MIN}/ELEN is reserved, but implementations can set vill if they do not support these configurations.

Having said that, I'm puzzled why it was `sew > std::min(vflmul, 1.0f) * elen` instead of `sew > vflmul * elen`.

**Comment by huangxiansheng123** [comment]
The sew here does not seem to be the smallest and does not match the SEW_{MIN} here in spec
